packages/keycloak-user-management/src/components/UserList/index.tsx

```diff
@@ -170,33 +173,25 @@ const UserList = (props: UserListTypes): JSX.Element => {
             </Space>
           </div>
           <Space>
-            <Table
+            <TableLayout
               columns={getTableColumns(
                 removeKeycloakUsersCreator,
                 keycloakBaseURL,
                 isLoadingCallback,
                 extraData,
                 sortedInfo
               )}
-              dataSource={tableData}
+              datasource={tableData}
               pagination={{
-                showQuickJumper: true,
-                showSizeChanger: true,
-                defaultPageSize: usersPageSize,
-                pageSize: pageProps.pageSize,
-                current: isSearchActive ? 1 : pageProps.currentPage,
-                onChange: (page: number, pageSize: number | undefined) => {
-                  setPageProps({
-                    currentPage: page,
-                    pageSize: pageSize,
-                  });
-                  setIsLoading(true);
-                },
+                ...defaults.pagination,
+                current: page.current,
+                pageSize: page.pageSize,
                 total: isSearchActive ? keycloakUsers.length : usersCount,
-                pageSizeOptions: ['5', '10', '20', '50', '100'],
               }}
-              onChange={(_: Dictionary, __: Dictionary, sorter: Dictionary) => {
+              onChange={(pagination, __, sorter) => {
+                setPage({ current: pagination.current ?? 1, pageSize: pagination.pageSize });
```

Comment:
refactor the pageSize value to also use the `userPageSize` prop incase `pagination.pageSize` is undefined

Reply:
but what if the user update the pagesize it wount update .. and i dont think it can be undefined as we are already giving it a page size by default and even if the user change it he can only change it to some value it can never be undefined

Reply:
yes you can refactor it to `pageSize: pagination.pageSize ?? usersPageSize`